src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java

```diff
@@ -77,6 +90,146 @@ default Uni<FindResponse> findDocument(
+   * @param errorLimit - Count of record on which system to error out, this will be (maximum read
+   *     count for sort + 1)
+   * @return
+   */
+  default Uni<FindResponse> findOrderDocument(
+      QueryExecutor queryExecutor,
+      QueryOuterClass.Query query,
+      int pageSize,
+      ObjectMapper objectMapper,
+      Comparator<ReadDocument> comparator,
+      int numberOfOrderByColumn,
+      int skip,
+      int limit,
+      int errorLimit) {
+    final AtomicInteger documentCounter = new AtomicInteger(0);
+    MinMaxPriorityQueue<ReadDocument> sortedData =
+        MinMaxPriorityQueue.orderedBy(comparator).maximumSize(skip + limit).create();
+    final JsonNodeFactory nodeFactory = objectMapper.getNodeFactory();
+    return Multi.createBy()
+        .repeating()
+        .uni(
+            () -> new AtomicReference<String>(null),
+            stateRef -> {
+              return queryExecutor.executeRead(
+                  query, Optional.ofNullable(stateRef.get()), pageSize);
+            })
+        // Read document while pagingState exists, limit for read is set at updateLimit +1
+        .whilst(resultSet -> extractPagingStateFromResultSet(resultSet) != null)
+        .onItem()
+        .transformToUniAndMerge(
+            resultSet -> {
+              Iterator<QueryOuterClass.Row> rowIterator =
+                  resultSet.getRowsList().stream().iterator();
+              int remaining = resultSet.getRowsCount();
+              documentCounter.addAndGet(remaining);
+              while (--remaining >= 0 && rowIterator.hasNext()) {
+                ReadDocument document = null;
+                QueryOuterClass.Row row = rowIterator.next();
+                List<JsonNode> sortValues = new ArrayList<>(numberOfOrderByColumn);
+                for (int sortColumnCount = 0;
+                    sortColumnCount < numberOfOrderByColumn;
+                    sortColumnCount++) {
+                  int columnCounter =
+                      SORTED_DATA_COLUMNS + ((sortColumnCount) * SORT_INDEX_COLUMNS_SIZE);
+                  QueryOuterClass.Value value = row.getValues(columnCounter);
+                  if (!value.hasNull()) {
+                    sortValues.add(nodeFactory.textNode(Values.string(value)));
+                    continue;
+                  }
+                  columnCounter++;
+                  value = row.getValues(columnCounter);
+                  if (!value.hasNull()) {
+                    sortValues.add(nodeFactory.numberNode(Values.decimal(value)));
+                    continue;
+                  }
+                  columnCounter++;
+                  value = row.getValues(columnCounter);
+                  if (!value.hasNull()) {
+                    sortValues.add(nodeFactory.booleanNode(Values.bool(value)));
+                    continue;
+                  }
+                  columnCounter++;
+                  value = row.getValues(columnCounter);
+                  if (!value.hasNull()) {
```

Comment:
Since `query_null_values` is a set, not map, what kind of value will be queried from it?

Reply:
In case of set or list type cassandra will return the entry name came in request if it contains the value